basic.go

```diff
@@ -11,18 +11,18 @@ import (
 ///////////////////////
 // BUS
 
-type bus struct {
+type BasicBus struct {
```

Comment:
Make the struct not constructive by making it private:

`s/BasicBus/basicBus/`